domain/domain.go

```diff
@@ -1702,24 +1702,24 @@ func (do *Domain) distTaskFrameworkLoop(ctx context.Context, taskManager *storag
 		logutil.BgLogger().Info("dist task scheduler stopped")
 	}()
 
-	var dispatch dispatcher.Dispatch
+	var dispatcherPool *dispatcher.Pool
 	startDispatchIfNeeded := func() {
-		if dispatch != nil {
+		if dispatcherPool != nil {
```

Comment:
Since dispatcherPool is a point type, and here will check if dispatcherPool is nil. So you need init point value.